internal/db/profiles_test.go

```diff
@@ -179,6 +191,101 @@ func createTestRandomEntities(t *testing.T) *testRandomEntities {
 	}
 }
 
+func TestProfileLabels(t *testing.T) {
+	t.Parallel()
+
+	randomEntities := createTestRandomEntities(t)
+
+	health1 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{"stacklok:health"})
+	require.NotEmpty(t, health1)
+	fmt.Println(health1.Name)
+	health2 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{"stacklok:health", "obsolete"})
+	require.NotEmpty(t, health2)
+	fmt.Println(health2)
+	p1 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{})
+	require.NotEmpty(t, p1)
+	fmt.Println(p1)
+	p2 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{})
+	require.NotEmpty(t, p2)
+	fmt.Println(p2)
+
+	tests := []struct {
+		name          string
+		includeLabels []string
+		excludeLabels []string
+		expectedNames []string
+	}{
+		{
+			name:          "list all profiles",
+			includeLabels: []string{"*"},
+			expectedNames: []string{health1.Name, health2.Name, p1.Name, p2.Name},
+		},
+		{
+			name:          "list profiles with no labels",
+			includeLabels: []string{},
+			expectedNames: []string{p1.Name, p2.Name},
+		},
+		{
+			name:          "list profiles with no labels using nil - default ",
+			includeLabels: nil,
+			expectedNames: []string{p1.Name, p2.Name},
+		},
+		{
+			name:          "list obsolete profile",
+			includeLabels: []string{"obsolete"},
+			expectedNames: []string{health2.Name},
+		},
+		{
+			name:          "list health profile",
+			includeLabels: []string{"stacklok:health"},
+			expectedNames: []string{health1.Name, health2.Name},
+		},
+		{
+			name:          "list health and obsolete profile",
+			includeLabels: []string{"stacklok:health", "obsolete"},
```

Comment:
wait, do labels work as an `OR`? I would have expected them to work as an `AND` similar to k8s. That is, listing profiles with includeLabels being `stacklok:health,obsolete` would list only profiles that contain both labels.

Reply:
Would be ideal to support something as follows: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement

Reply:
> wait, do labels work as an `OR`? I would have expected them to work as an `AND` similar to k8s. That is, listing profiles with includeLabels being `stacklok:health,obsolete` would list only profiles that contain both labels.

No, it's an AND, check the expected result:
```go
	{
			name:          "list health and obsolete profile",
			includeLabels: []string{"stacklok:health", "obsolete"},
			expectedNames: []string{health2.Name},
	},
```
The `health2` profile is:
```go
	health1 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{"stacklok:health"})
	require.NotEmpty(t, health1)
	fmt.Println(health1.Name)
	health2 := createRandomProfile(t, randomEntities.prov, randomEntities.proj.ID, []string{"stacklok:health", "obsolete"})
	require.NotEmpty(t, health2)
```
so `health1` doesn't match, only `health2`. On the SQL side this is implemented using the [array operator](https://www.postgresql.org/docs/current/functions-array.html#FUNCTIONS-ARRAY):
```sql
    (COALESCE(cardinality(sqlc.arg(include_labels)::TEXT[]), 0) > 0 AND profiles.labels @> sqlc.arg(include_labels)::TEXT[])
```
which means that every element in the include_labels argument must be a member of the profile labels.

I will add another profile with just `obsolete` to test this better.

Reply:
> Would be ideal to support something as follows: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement

Our labels unlike k8s labels don't contain values, can you give me some examples of queries you'd like to see added?

Reply:
oh, sorry, I think the name of the testcase was very confusing (I admit I use copilot to generate the names..).
I'll fix that and also remove the stray `fmt.Println()`